dali/operators/reader/loader/webdataset/tar_utils.cc

```diff
@@ -140,15 +139,32 @@ bool TarArchive::NextFile() {
   if (eof_) {
     return false;
   }
-  Skip(RoundToBlockSize(filesize_) - readoffset_);
-  eof_ = ParseHeader();
+
+  if (stream_->Tell() + RoundToBlockSize(filesize_) - readoffset_ >= stream_->Size()) {
+    SetEof();
+    return false;
+  }
+  stream_->Seek(stream_->Tell() + RoundToBlockSize(filesize_) - readoffset_);
+
+  ParseHeader();
   return !eof_;
 }
 
 bool TarArchive::EndOfArchive() const {
   return eof_;
 }
 
+void TarArchive::Seek(size_t offset) {
```

Comment:
```suggestion
void TarArchive::Seek(int64_t offset) {
```
For consistency with FileStream?

Reply:
done